metrics/scaling/k8s_scale_rapid_nc.sh

```diff
@@ -0,0 +1,337 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+# Pull in some common, useful, items
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+LABELVALUE=${LABELVALUE:-rapid_nc}
+source "${SCRIPT_PATH}/../lib/common.bash"
+source "${SCRIPT_PATH}/common.bash"
+source "${SCRIPT_PATH}/../collectd/collectd.bash"
+
+SMF_USE_COLLECTD=true
+
+# Network latency test parameters:
+# number of requests to be sent after each scaling step
+nc_reqs=${nc_reqs:-1000}
+# length of each request [bytes]
+nc_req_msg_len=${nc_req_msg_len:-1000}
+# port that request servers listen to in pods
+nc_port=33101
+# request message
+nc_req_msg=$(head -c $nc_req_msg_len /dev/zero | tr  '\0' 'x')
+nc_percentiles=(0 1 5 25 50 75 95 99 100)
+
+pod_command="[\"nc\", \"-lk\", \"-p\", \"${nc_port}\", \"-e\", \"/bin/sh\", \"-c\", \"/bin/echo \${EPOCHREALTIME/./}; /bin/cat; /bin/echo \${EPOCHREALTIME/./}\"]"
+
+# Set some default metrics env vars
+TEST_ARGS="runtime=${RUNTIME}"
+TEST_NAME="k8s rapid nc"
+
+# $1 is the launch time in seconds this pod/container took to start up.
+# $2 is the number of pod/containers under test
+grab_stats() {
+	local launch_time_ms=$1
+	local n_pods=$2
+	shift ; shift
+	local latency_percentiles=($@) # array of percentiles
+	local cpu_idle=()
+	local mem_free=()
+	local total_mem_used=0
+
+	info "And grab some stats"
+
+	local date_json="$(cat << EOF
+			"date": {
+				"ns": $(date +%s%N),
+				"Date": "$(date -u +"%Y-%m-%dT%T.%3N")"
+			}
+EOF
+	)"
+	metrics_json_add_array_fragment "$date_json"
+
+	local pods_json="$(cat << EOF
+			"n_pods": {
+				"Result": ${n_pods},
+				"Units" : "int"
+			}
+EOF
+	)"
+	metrics_json_add_array_fragment "$pods_json"
+
+	local launch_json="$(cat << EOF
+			"launch_time": {
+				"Result": $launch_time_ms,
+				"Units" : "ms"
+			}
+EOF
+	)"
+	metrics_json_add_array_fragment "$launch_json"
+
+	local latency_json="$(cat << EOF
+			"latency_time": {
+				"Percentiles": [$(IFS=, ; echo "${latency_percentiles[*]}")],
+				"Result": ${latency_percentiles[$(( ${#latency_percentiles[@]} / 2 ))]},
```

Comment:
Nitpick - we might want to rename `Result` to something like `Midpoint` to reflect the actual value it holds.